### ISSUE TITLE
ng g i allows the user to mistakenly create a folder which cannot be deleted 

### ISSUE THREAD

**Issue by williamBurgeson** [angular/angular-cli#6342]
HIGH PRIORITY - BUG

I wanted to create an interface file and used the command "ng g i shared/registration/registration.interface /dr" to dry run it, inadvertently using / instead of - for the dry run.

Instead of returning an error, which it should have done, it created he following file:-
shared\registration\registration.\dr.ts

The shared\registration folder already existed, however I don't think that has any bearing on the situation.

I was unable to delete this newly created "registration." folder, or do anything with it, or its containing folder, and when attempting to delete in windows explorer I was presented with the error 0x80004005, which only after much searching on google I managed to find the solution of running del "\\?\C:\[...]\shared\registration".

Obviously this is a serious bug in Windows (10), however Angular-CLI should protect the user from this sort of situation, probably by double-checking that none of the elements of the file path created contain a . (dot) character at the end

**Comment by sumitarora** [comment]
@williamBurgeson Was the dev server running when you tried to delete the folder?

**Comment by williamBurgeson** [comment]
If you're referring to ng serve, or likewise, then no; it's a .net core project so I do ng build and press play in visual studio.
Bearing in mind it can't be deleted either in windows explorer or powershell (I didn't think to try command line).
Note the face that I'm calling the file xxx.interface, and the ng cli seems to strip off the "interface" part of the filename.

**Comment by mgechev** [comment]
Are you still able to reproduce this issue with the latest version of the CLI?

**Comment by mgechev** [comment]
Closing the issue due to inactivity. If the problem still exists in the latest version of the CLI, please open a new issue following the template and providing a minimal demo where we can reproduce it. Thanks! :-)
